### PR TITLE
Mail credentials

### PR DESCRIPTION
**Pull request by normanbos**
The new feature branch let's the user input their own email credentials and stores them in local storage.

### CODE REVIEW

File: src/components/MailCredentialsModal.js [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+            <CancelButton onClick={closeModal} />
+            <SaveButton type="submit" form="cardForm" onClick={handleSave} />
+          </CardHeaderNav>
+        </CardHeader>
+        <p>
+          Um die Benachrichtungsfunktion verwenden zu können, trage deine
+          E-Mail-Zugangsdaten hier ein:
+        </p>
+        <CardForm id="cardForm">
+          <Label htmlFor="host">
```
Comment (arisslin) [inline src/components/MailCredentialsModal.js]: You can put this code 
```
<small>
  <i>SMTP-Host</i>
</small>
```
into your Label component and transfer the text content with a prop.

File: src/components/MailCredentialsModal.js [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+  function handleItemChange(event) {
+    setMailCreds({
+      ...mailCreds,
+      [event.target.name]: event.target.value,
+    })
+  }
+}
+
+export const StyledModal = {
```
Comment (arisslin) [inline src/components/MailCredentialsModal.js]: Maybee use styled components.

File: src/components/ReminderModal.js [hunk 3]
```diff
@@ -1,14 +1,13 @@
-import React, { useState } from 'react'
+import React from 'react'
 import { ReminderMailer } from './reminderMailer'
 import Modal from 'react-modal'
 
-//Modal.setAppElement(document.getElementById('root'))
-
 export default function ReminderModal({
```
Comment (arisslin) [inline src/components/ReminderModal.js]: Think about proptypes.

File: src/components/buttons.js [hunk 4]
```diff
@@ -1,6 +1,6 @@
 import React from 'react'
 import { FaEdit, FaReply, FaSave, FaTrashAlt } from 'react-icons/fa'
-import { IoIosAddCircle } from 'react-icons/io'
+import { IoIosAddCircle, IoMdSettings } from 'react-icons/io'
 import styled from 'styled-components/macro'
 import { ReactComponent as Icon } from '../img/mailalertplaceholder.svg'
```
Comment (arisslin) [inline src/components/buttons.js]: Think about proptypes for all the button components in this file.

File: src/components/buttons.js [hunk 5]
```diff
@@ -24,6 +24,23 @@ export function MailAlertButton({ onClick }) {
   )
 }
 
+export function SettingsButton({ onClick }) {
+  return (
+    <Button
+      onClick={onClick}
+      css={`
```
Comment (arisslin) [inline src/components/buttons.js]: Why not put this into the styled.components below?

File: src/components/formCreateCard.js [hunk 6]
```diff
@@ -13,7 +13,7 @@ export function FormCreateCard({ onSubmit, cancelHandle }) {
     duedate: '',
     id: v4(),
   })
-
+  console.log(itemState)
```
Comment (arisslin) [inline src/components/formCreateCard.js]: Delete the log before merging into master branch.

File: src/components/reminderMailer.js [hunk 7]
```diff
@@ -5,7 +5,13 @@ import { ReminderButton } from './buttons'
 import { CardFooter } from './Card'
 import CloseModalCountdown from './closeModalCountdown'
 
-export function ReminderMailer({ item, borrower, contact, closeModal }) {
+export function ReminderMailer({
```
Comment (arisslin) [inline src/components/reminderMailer.js]: Think about proptypes.

File: src/components/MailCredentialsModal.js (hunk 2 above)
Comment (normanbos) [reply src/components/MailCredentialsModal.js]: I think the react-modal is not compatible with styled components

**Pull request merged**